lib/patreon/utils/jsonapi/url_util.rb

```diff
@@ -5,13 +5,15 @@ module Patreon
   module Utils
     module JSONAPI
       class URLUtil
-        def self.build_url(url, includes=nil, fields=nil)
+        def self.build_url(url, includes=nil, fields=nil, count=(count_not_passed = true; nil), cursor=nil)
           parsed_url = URI.parse(url)
           params = parsed_url.query ? Rack::Utils.parse_query(parsed_url.query) : {}
           params['include'] = joined_or_null(includes) if includes
           fields.each do |name, val|
             params["fields[#{name}]"] = val
           end if fields
+          params["page[count]"] = (count || 10) unless count_not_passed
```

Comment:
This desugars to:
```ruby
if not count_not_passed
  params["page[count]"] = (count || 10)
end
```

When this method is:
- Called with `count` set to `3`, then `count_not_passed == false`, `count == 3`. This sets `params["page[count]"]` to `3 || 10`, which evaluates to `3`.
- Called with `count` set to `nil`, then `count_not_passed == false`, `count == nil`. This sets `params["page[count]"]` to `nil || 10`, which evaluates to `10`.
- Called without `count`, then `count_not_passed == true`, `count == nil`. This doesn't set `params["page[count]"]` at all.

It looks like the intent is to provide three choices?
1. Use the `count` specified.
1. Use default `count`.
1. Omit `count`.

If this triple-choice is to stay, I recommend turning the rest of the optional arguments into keyword args. Depending on the minimum Ruby version supported, we could either use real Ruby keyword args, or simulate them with a hash, as is a long-time traditional in older versions of Ruby.

Reply:
Great catch and great explanation! Looking at the python lib I don't even think we need to provide a default. I will update the logic to:

1. Use the count specified.
2. Omit count.